wordpress-mobile/WordPress-iOS
# Update Auth to use Google login refactor.

Ref #https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/282
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/284

This uses the Auth branch to utilize the Google login refactor.  To note, there should be _no_ functional or visible changes. Testing is to verify nothing is broken.

To test:

---
**Test Case 1: Login with Google account**
1. Have an account that was created via `Sign up with Google` (i.e so it is flagged as a social account).
2. Go to `Log In` > `Continue with Google`.
3. Walk through the Google account authentication.
4. Verify the Login Epilogue is displayed.

![epilogue](https://user-images.githubusercontent.com/1816888/82606619-9c5be080-9b74-11ea-803d-6720841eeb84.png)

---
**Test Case 2: Login with Google account that has 2FA enabled on the WP account**
1. Have the same kind of account as in Test Case 1, but with 2FA enabled on the WordPress account.
2. Go to `Log In` > `Continue with Google`.
3. Walk through the Google account authentication.
4. Verify the 2FA code prompt is displayed.

![2fa](https://user-images.githubusercontent.com/1816888/82606638-a4b41b80-9b74-11ea-93ed-416ab9c33b4f.png)

---
**Test Case 3: Login with Google using an existing WP account**
1. Have an account that was created via `Sign up with Email` (i.e so it is flagged as a WP account).
2. Go to `Log In` > `Continue with Google`.
3. Walk through the Google account authentication.
4. Verify the WP email prompt is displayed.

![existing_wp_account](https://user-images.githubusercontent.com/1816888/82606672-b5649180-9b74-11ea-90f4-143fe06bdcb1.png)

---
**Test Case 4: Login with Google fails**
1. Have the same kind of account as in Test Case 2.
2. Go to `Log In` > `Continue with Google`.
3. Walk through the Google account authentication.
3. Immediately log out and attempt to login again via Google.
4. Verify the `Unable to Connect` view is displayed.

![fail](https://user-images.githubusercontent.com/1816888/82606702-be556300-9b74-11ea-9e8e-b7ae123ff486.png)

---
PR submission checklist:
